lib/manager/sbt/readme.md

```diff
@@ -0,0 +1,245 @@
+
+---
+
+#### What [fileMatch](https://renovatebot.com/docs/configuration-options/#filematch) pattern(s) should be used?
+
+`['build\\.sbt', 'plugins\\.sbt']`
+
+Note that `plugins.sbt` file is usually in a `project/` subdirectory.
+
+---
+
+#### Is it likely that many users would need to extend this pattern for custom file names?
+
+Not likely.
+
+---
+
+#### Is the fileMatch pattern likely to get many "false hits" for files that have nothing to do with package management?
+
+Not likely. The extension `.sbt` should not be used for other files.
+
+## Parsing and Extraction
+
+#### Can package files have "local" links to each other that need to be resolved?
+
+No
+
+#### Is there reason why package files need to be parsed together (in serial) instead of independently?
+
+No
+
+---
+
+#### What format/syntax is the package file in? e.g. JSON, TOML, custom?
+
+Scala syntax
+
+---
+
+#### How do you suggest parsing the file? Using an off-the-shelf parser, using regex, or can it be custom-parsed line by line?
+
+I think reading the files with Scala would be the easier. Due to the fact that the files are plain Scala code, the
+dependencies can be declared in multiple ways.
+
+But, I guess there is two most commonly used way to declare dependencies and using regex could work for these formats.
+
+Another option might be to use ScalaJS. I don't know it but it has same syntax as Scala and can be read/used within Javascript code.
+
+If we can use Scala or ScalaJS, the idea would be to execute the `build.sbt` or `plugins.sbt` file and work on the `libraryDependencies` or `dependencyOverrides` variables.
+
+_Update (thanks to @ChristianMurphy)_: https://github.com/scalameta/scalameta running on http://www.scala-js.org could be a way to parse Scala without depending on a JDK.
+E.G. https://astexplorer.net/#/gist/027dce11e6927b4ad39ea097ce6289b9/ea4d1048f32063a71d727d01178ae6d01087a62f
+
+---
+
+#### Does the package file structure distinguish between different "types" of dependencies? e.g. production dependencies, dev dependencies, etc?
+
+Dependencies can have a "scope" like `Test` or `Provided`.
+
+Renovate shouldn't care of scopes: they should be kept as is but they don't have any impact on the "new dependency resolution".
+
+---
+
+#### List all the sources/syntaxes of dependencies that can be extracted:
+
+##### build.sbt
+
+Most common syntaxes:
+
+```scala
+// For Scala dependency (notice the double %%)
+libraryDependencies += "com.typesafe.play" %% "play-ws" % "2.6.10"
+// For Java dependency
+libraryDependencies += "org.mockito" % "mockito-all" % "1.10.18"
+
+// With a scope
+libraryDependencies += "org.mockito" % "mockito-all" % "1.10.18" % Test
+
+// Multiple dependencies at the same time
+libraryDependencies ++= Seq(
+  "com.typesafe.play" %% "play-ws" % "2.6.10",
+  "org.mockito" % "mockito-all" % "1.10.18"
+)
+
+// Defining some dependency overriding (same as for build.sbt)
+dependencyOverrides += "com.google.guava" % "guava" % "23.0"
```

Comment:
What is this declaration doing? e.g. what is it overriding? Is guava a nested dependency?

Reply:
Added some comments few line above. 
Dependency overriding is for transitive/nested dependencies. It will force a version in case all the libraries dependencies don't depend on the same version of this (overriden) library.
See also: https://www.scala-sbt.org/1.x/docs/Library-Management.html#Overriding+a+version